docs/sources/configuration/integrations/gcp-exporter-config.md

```diff
@@ -0,0 +1,186 @@
+  # Using pubsub metrics (https://cloud.google.com/monitoring/api/metrics_gcp#gcp-pubsub) as an example
+  # all metrics
+  #   - pubsub.googleapis.com/
+  # all snapshot specific metrics 
+  #   - pubsub.googleapis.com/snapshot
+  # all snapshot specific metrics and a few subscription metrics
+  #   - pubsub.googleapis.com/snapshot
+  #   - pubsub.googleapis.com/subscription/num_undelivered_messages
+  #   - pubsub.googleapis.com/subscription/oldest_unacked_message_age
+  metrics_prefixes:
+    [ - <string> ... ]
+
+  # Optional: Used to further refine the resources you would like to collect metrics from
+  # The structure for these filters is <targeted_metric_prefix>:<filter_query>
+  # The `targeted_metric_prefix` is used to ensure the filter is only applied to the metric_prefix(es) where it makes sense.
+  #   It does not explicitly have to match a value from `metric_prefixes` but the `targeted_metric_prefix` must be at least a
+  #   prefix to one or more `metric_prefixes`
+  #   Example: 
+  #     metrics_prefixes = pubsub.googleapis.com/snapshot, pubsub.googleapis.com/subscription/num_undelivered_messages
+  #     targeted_metric_prefix options would be 
+  #       pubsub.googleapis.com (apply to all defined prefixes)
+  #       pubsub.googleapis.com/snapshot (apply to only snapshot metrics)
+  #       pubsub.googleapis.com/subscription (apply to only subscription metrics)
+  #       pubsub.googleapis.com/subscription/num_undelivered_messages (apply to only the specific subscription metric)
+  # The `filter_query` will be applied to a final metrics API query when querying for metric data
+  #   You can read more about the metric API filter options in GCPs documentation https://cloud.google.com/monitoring/api/v3/filters
+  #   The final query sent to the metrics API already includes filters for project and metric type. Each applicable `filter_query`
+  #   will be appended to the query with an AND
+  extra_filters:
+    [ - <string> ... ]
+
+  # Optional: the time range used when querying for metrics
+  # Most of the time the default works perfectly fine. Most documented metrics include a comments of the form 
+  #   `Sampled every X seconds. After sampling, data is not visible for up to Y seconds.`
+  #   As longs as your `request_interval` is >= `Y` you should have no issues
+  #   Consider using `ingest_delay` if you would like this to be done programmatically or are gathering slower moving metrics
+  [request_interval: <duration> | default = "5m"]
+
+  # Optional: when enabled this automatically adjusts the time range used when querying for metrics backwards based on
+  #   the metadata GCP has published for how long the data can take to be ingested. You can see the values for this in 
+  #   documented metrics as `After sampling, data is not visible for up to Y seconds.`
+  # Since GCPs ingestion delay is an "at worst" this is off be default to ensure data is gathered as soon as it's available  
```

Comment:
```suggestion
  # Optional: When enabled this automatically adjusts the time range used when querying for metrics backwards based on
  #   the metadata GCP has published for how long the data can take to be ingested. You can see the values for this in 
  #   documented metrics as `After sampling, data is not visible for up to Y seconds.`
  # Since GCPs ingestion delay is an "at worst," this is off by default to ensure data is gathered as soon as it's available.
```